libs/SmartSync/SmartSync/Classes/Target/SFParentChildrenSyncDownTarget.m

```diff
@@ -159,20 +159,25 @@ - (NSString*) getQueryToRun:(long long)maxTimeStamp {
 
 - (void)cleanGhosts:(SFSmartSyncSyncManager *)syncManager
            soupName:(NSString *)soupName
+             syncId:(NSNumber *)syncId
          errorBlock:(SFSyncDownTargetFetchErrorBlock)errorBlock
       completeBlock:(SFSyncDownTargetFetchCompleteBlock)completeBlock {
 
     // Taking care of ghost parents
     [super cleanGhosts:syncManager
               soupName:soupName
+                syncId:syncId
             errorBlock:errorBlock
          completeBlock:^(NSArray *localIdsArr) {
 
              // Taking care of ghost children
 
              // NB: ParentChildrenSyncDownTarget's getNonDirtyRecordIdsSql does a join between parent and children soups
              // We only want to look at the children soup, so using SoqlSyncDownTarget's getNonDirtyRecordIdsSql
-             NSMutableOrderedSet* localChildrenIds = [[self getIdsWithQuery:[super getNonDirtyRecordIdsSql:self.childrenInfo.soupName idField:self.childrenInfo.idFieldName] syncManager:syncManager] mutableCopy];
+             NSMutableOrderedSet *localChildrenIds = [[self getIdsWithQuery:[super getNonDirtyRecordIdsSql:self.childrenInfo.soupName
```

Comment:
I think `self` here also will get retained since it's inside the block.

Reply:
I wonder if we are not doing the strongself/weakself sometimes when we don't need to.

If self doesn't keep a reference to the block there is no chance of retain cycle.
When only using blocks inline, we shouldn't have to use weak self. (see https://stackoverflow.com/questions/13194613/is-it-necessary-to-use-weak-references-to-self-always-inside-blocks/13194818#13194818).

Also since, we have the `-Wimplicit-retain-self` flag, wouldn't the compiler warn us?